doc/source/reference/random/upgrading-pcg64.rst

```diff
@@ -0,0 +1,153 @@
+One of the known properties of an LCG is that different increments create
+*distinct* streams, but with a known relationship. Each LCG has an orbit that
+traverses all :math:`2^{128}` different 128-bit states. Two LCGs with different
+increments are related in that one can "rotate" the orbit of the first LCG
+(advance it by a number of steps that we can compute from the two increments)
+such that then both LCGs will always then have the same state, up to an
+additive constant and maybe an inversion of the bits. If you then iterate both
+streams in lockstep, then the states will *always* remain related by that same
+additive constant (and the inversion, if present). Recall that `PCG64` is
+constructed from both a transition function (the LCG) and an output function.
+It was expected that the scrambling effect of the output function would have
+been strong enough to make the distinct streams practically independent (i.e.
+"passing the ``PractRand`` tests") unless the two increments were
+pathologically related to each other (e.g. 1 and 3). The output function XSL-RR
+of the then-standard PCG algorithm that we implemented in `PCG64` turns out to
+be too weak to cover up for the 58-bit collision of the underlying LCG that we
+described above. For any given pair of increments, the size of the "colliding"
+space of states is the same, so for this weakness, the extra distinctness
+provided by the increments does not translate into extra protection from
+statistical correlations that ``PractRand`` can detect.
+
+Fortunately, strengthening the output function is able to correct this weakness
+and *does* turn the extra distinctness provided by differing increments into
+additional protection from these low-bit collisions. To the `PCG author's
+credit <https://github.com/numpy/numpy/issues/13635#issuecomment-506088698>`_,
+she had developed a stronger output function in response to related discussions
+during the long birth of the new `BitGenerator` system. We NumPy developers
+(mostly me, Robert Kern, to be clear where the blame lies) chose to be
+"conservative" and use the XSL-RR variant that had undergone a longer period of
+testing at that time. The DXSM output function adopts a "xorshift-multiply"
+construction used in strong integer hashes that has much better avalanche
+properties than the XSL-RR output function. While there are "pathological"
+pairs of increments that induce "bad" additive constants that relate the two
+streams, the vast majority of pairs induce "good" additive constants that make
+the merely-distinct streams of LCG states into practically-independent output
+streams. Indeed, now the claim I once made about `PCG64` is actually true of
+`PCG64DXSM`: collisions are possible, but both streams have to simultaneously
+be both "close" in the 128 bit state space *and* "close" in the 127-bit
+increment space, so the negligible chance of colliding in the 128-bit internal
+`SeedSequence` pool is more likely. The DXSM output function is more
```

Comment:
"pool is less likely"?

Reply:
No. It is more likely that you will observe a collision in the `SeedSequence` pool (creating completely identical seeds) than to see one of the poorly-correlated pairs of distinct streams.

Reply:
Changing the verb to "would be" would help (and be more consistent with the other verbiage we have on the subject).

Reply:
Does the `SeedSequence` affect both increment and state, or just the state?

Reply:
Both increment and state. During the mega-thread, we decided that was the best design, especially once we got strong seeding with `SeedSequence`.

Reply:
> Both increment and state

Does this make the seed sequences differ between bit-generators or is the increment derived from the state?

Reply:
Let me back up a bit to where I suspect the disconnect is. We'll see if I guessed right.

`SeedSequence` takes an arbitrary sequence of (arbitrary-sized) integers and hashes them down into an internal pool of bits. The default pool size is 128 bits. The `BitGenerator` then asks the `SeedSequence` for however many bits it needs to initialize itself. For `MT19937`, that's a plump 624 `uint32`s. For the `PCG64`s, that's the relatively-svelte 128-bit state and 127-bit increment (added together and rounded up to an even 256 bits total). The `SeedSequence` uses what amounts to a slow, carefully constructed PRNG to generate however many bits are requested. It's not just copying its internal pool of 128 bits into the `BitGenerator` state.

The claim I am making here is that the odds that two distinct `SeedSequence`s generate a pair of distinct, bad `PCG64DXSM` instances is less than the odds that two distinct user inputs hashed down to the same 128-bit internal pool. And the odds of that 128-bit collision are ignorably tiny.